src/lib/communication/DY.Lib.Communication.Core.Invariants.fst

```diff
@@ -128,34 +129,29 @@ let default_comm_higher_layer_event_preds (a:Type) {| parseable_serializeable by
 }
 
 #push-options "--ifuel 1 --fuel 0"
-let event_predicate_communication_layer 
+let event_predicate_communication_layer
   {|cinvs:crypto_invariants|}
   (#a:Type) {| parseable_serializeable bytes a |}
-  (higher_layer_preds:comm_higher_layer_event_preds a) : 
+  (higher_layer_preds:comm_higher_layer_event_preds a) :
   event_predicate communication_event =
   fun tr prin e ->
     (match e with
     | CommConfSendMsg sender receiver payload -> (
       is_knowable_by (comm_label sender receiver) tr payload /\
-      (match parse a payload with
-      | None -> False
-      | Some payload_p -> higher_layer_preds.send_conf tr sender receiver payload_p
-      )
+      parse_and_pred (higher_layer_preds.send_conf tr sender receiver) payload
     )
     | CommConfReceiveMsg receiver payload -> (
-      exists sender.
-        (
-          event_triggered tr sender (CommConfSendMsg sender receiver payload) \/
-          is_publishable tr payload
-        )
+      Some? (parse a payload) /\
+      (
+        (exists sender. event_triggered tr sender (CommConfSendMsg sender receiver payload)) \/
+        is_publishable tr payload
+      )
     )
     | CommAuthSendMsg sender payload -> (
-      (match parse a payload with
-      | None -> False
-      | Some payload_p -> higher_layer_preds.send_auth tr sender payload_p
-      )
+      parse_and_pred (higher_layer_preds.send_auth tr sender) payload
     )
     | CommAuthReceiveMsg sender receiver payload -> (
+      Some? (parse a payload) /\
```

Comment:
Same as above

Reply:
This has the same purpose as above, to make it so that the receive event contains the same information that we get from calling the receive function (including that the payload parses successfully, without which, the function fails and does not create this event). Here, it doesn't have as obvious of a role, but if at some point in the future, we want to make use of knowing that a message was received at a time other than when it was just received, it still seems useful to have the receive function and the receive event give the same strength of guarantee.